sdk/eventhub/azure-eventhub/azure/eventhub/aio/_eventprocessor/event_processor.py

```diff
@@ -201,14 +207,19 @@ def _create_tasks_for_claimed_ownership(
                 await self._process_error(partition_context, err)
 
     async def _on_event_received(
-        self, partition_context: PartitionContext, event: EventData
+        self, partition_context: PartitionContext, event: Union[Optional[EventData], List[EventData]]
     ) -> None:
-        with self._context(event):
-            if self._track_last_enqueued_event_properties:
-                partition_context._last_received_event = (  # pylint: disable=protected-access
-                    event
-                )
-            await self._event_handler(partition_context, event)
+        if event:
+            # pylint: disable=protected-access
+            with self._context(event):
+                if event:
```

Comment:
We already have the `if event` check on line 212

Reply:
Re-phrased the code.